test/scripts/lib/utils.py

```
@@ -0,0 +1,121 @@
+
+    Required arguments:
+    l -- The iterable to be sorted.
+
+    """
+    convert = lambda text: int(text) if text.isdigit() else text
+    alphanum_key = lambda key: [convert(c) for c in re.split('([0-9]+)', key)]
+    return sorted(l, key = alphanum_key, reverse = reverse)
+
+def generate_og_alert(service_key, message, description, debug):
+    if debug:
+        log.warn(message, exception=description)
```

Ruff (0.15.6)

[error] Undefined name `log`

(F821)